Library/Homebrew/utils/lock.sh

```diff
@@ -0,0 +1,31 @@
+# create a lock using `flock (2)`. A name is required as first argument.
+# the lock will be automatically unlocked when the shell process quits.
+lock() {
+  local name="$1"
+  local lock_file="/tmp/homebrew-${name}.lock"
+  # open the lock file to file descriptor 3, so the shell process can hold the lock.
```

Comment:
`3` is fine: there just needs to be a comment explaining why it was chosen and not 0, 1, 2, 4, 800 and what happens if another process also uses `flock` on `3` at the same time.